src/main/java/duke/command/AddPatientCommand.java

```diff
@@ -11,21 +11,20 @@
 import duke.task.TaskManager;
 
 public class AddPatientCommand extends Command {
-    private String command;
-    public AddPatientCommand(String command) {
-        super();
-        this.command = command;
-    }
 
-    @Override
-    public void execute(PatientTaskList patientTask, TaskManager tasks, PatientManager patientList, Ui ui, PatientTaskStorage patientTaskStorage, TaskStorage taskStorage, PatientStorage patientStorage) throws DukeException {
-        Patient newPatient;
+    private Patient newPatient;
+
+    public AddPatientCommand(String[] patientInfo) throws DukeException {
+        super();
         try {
-            String commandArr[] = command.split("\\s+", 4);
-            newPatient = new Patient(commandArr[0], commandArr[1], commandArr[2], commandArr[3]);
+            this.newPatient = new Patient(patientInfo[0], patientInfo[1], patientInfo[2], patientInfo[3]);
         } catch (Exception e) {
             throw new DukeException("Please follow the format 'add patient <name> <NRIC> <Room> <remark>'. ");
         }
```

Comment:
Does it means that for the command that needs to check the validity of a specific patientInfo's column, the checking process will be executed in the constructor too?
Example such as `delete patient #<id>`

Reply:
When I altered the construction of the Patient object in the AddPatientCommand constructor, it was meant to streamline the calling process in the CommandManager.

If an object has a field that needs to be checked for validity, it will then need to be checked in the constructor of the specific Command class.

Reply:
Got it. It sounds much more reasonable than the previous approaches.